packages/cloud_firestore_odm/cloud_firestore_odm/lib/src/firestore_reference.dart

```diff
@@ -94,12 +96,13 @@ abstract class FirestoreCollectionReference<
   FirestoreDocumentReference doc([String? id]);
 }
 
-abstract class FirestoreDocumentSnapshot {
+abstract class FirestoreDocumentSnapshot<Model> {
   /// The reference for this document.
-  FirestoreDocumentReference get reference;
+  FirestoreDocumentReference<Model, FirestoreDocumentSnapshot<Model>>
+      get reference;
 
   /// The original [DocumentSnapshot] returned by Firebase.
-  DocumentSnapshot<Object?> get snapshot;
+  DocumentSnapshot<Model> get snapshot;
```

Comment:
Isn't it supposed to be nullable?
```suggestion
  DocumentSnapshot<Model?> get snapshot;
```

Reply:
No :) 

It was `Object?` before because it was unknown. But since it's now known, there's no reason to force it to be nullable.

What matters is that `DocumentSnapshot.data` returns a nullable value, which it does

Reply:
Okay nice thanks for the explanation ☺️